Xamarin.Forms.Core/Button.cs

```diff
@@ -239,6 +239,28 @@ void OnSourcePropertyChanging(ImageSource oldvalue, ImageSource newvalue)
 				oldvalue.SourceChanged -= OnSourceChanged;
 		}
 
+		// Windows 8.1/WP 8.1 have a different default behavior where if the BorderThickness is not explicitly set, it
+		// uses a BorderThickness of 2.5 on all sides on phone devices, and 2 otherwise. Because the BorderWidthProperty
+		// was otherwise set to 0d, it was not accounting for scenarios where a user did in fact want the button's border
+		// to be 0. If it is already 0 by default, the 8.1 Control's BorderThickness couldn't be properly compared. This
+		// private property to allow BorderWidthProperty to bind to for its correct default value and still uses 0 for
+		// the other platforms.
+		static double DefaultBorderWidth
+		{
+			get
+			{
+				double _value = 0d;
+				Device.OnPlatform(WinPhone: () =>
+				{
+					if (Device.Idiom == TargetIdiom.Phone)
+						_value = 2.5d;
```

Comment:
Sooo...why did we previously default to 3 if the defaults were actually 2 or 2.5?